services/qs-spark-service/src/main/java/ai/quantumics/api/service/impl/RuleServiceImpl.java

```diff
@@ -73,6 +79,120 @@ public ResponseEntity<Object> saveRule(RuleDetails ruleDetails, int userId, int
 		final Map<String, Object> response = new HashMap<>();
 		log.info("Invoking saveRule  API {}", ruleDetails.toString());
 		try {
+			String targetFeedName = null;
+			String targetFileName = null;
+			String targetBucketName = null;
+			String targetFilePattern = null;
+			if(ruleDetails.isSourceAndTarget()) {
+				DataSourceDetails targetDetails = ruleDetails.getTargetData();
+				if(targetDetails == null){
+					response.put("code", HttpStatus.SC_BAD_REQUEST);
+					response.put("message", "Target details can't be null");
+					return ResponseEntity.ok().body(response);
+				}
+                targetFilePattern = targetDetails.getFilePattern();
+				if(targetFilePattern == null){
+					response.put("code", HttpStatus.SC_BAD_REQUEST);
+					response.put("message", "Target file pattern can't be null");
+					return ResponseEntity.ok().body(response);
+				}
+				String filePath = targetDetails.getFilePath();
+				if(filePath == null){
+					response.put("code", HttpStatus.SC_BAD_REQUEST);
+					response.put("message", "Target file path can't be null");
+					return ResponseEntity.ok().body(response);
+				}
+				targetBucketName = targetDetails.getBucketName();
+				String targetFilePath = "s3://" + targetBucketName + "/" + filePath;
+
+				// Split strings
+				String[] targetFilePatternList = targetFilePattern.split("/");
+				String[] targetFilePathList = targetFilePath.split("/");
+				// Check if both arrays have the same size
+				if (targetFilePatternList.length != targetFilePathList.length) {
+					log.info(FILE_NOT_ALIGN);
+					response.put("code", HttpStatus.SC_BAD_REQUEST);
+					response.put("message", FILE_NOT_ALIGN);
+					return ResponseEntity.ok().body(response);
+				}
+				int targetFeedNameIndex = PatternUtils.getIndex(targetFilePatternList,FEED_NAME);
```

Comment:
we expect the 4 constants by default.. should not check for one by one and return. Check what are not available and return in message.